src/uu/du/src/du.rs

```diff
@@ -615,85 +589,80 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         "\n"
     };
 
-    let excludes = get_glob_ignore(&matches)?;
+    let excludes = build_exclude_patterns(&matches)?;
 
     let mut grand_total = 0;
     'loop_file: for path_string in files {
         // Skip if we don't want to ignore anything
         if !&excludes.is_empty() {
             for pattern in &excludes {
-                {
-                    if pattern.matches(path_string) {
-                        // if the directory is ignored, leave early
-                        if options.verbose {
-                            println!("{} ignored", path_string.quote());
-                        }
-                        continue 'loop_file;
+                if pattern.matches(path_string) {
+                    // if the directory is ignored, leave early
+                    if options.verbose {
+                        println!("{} ignored", path_string.quote());
                     }
+                    continue 'loop_file;
                 }
             }
         }
 
         let path = PathBuf::from(&path_string);
-        match Stat::new(path, &options) {
-            Ok(stat) => {
-                let mut inodes: HashSet<FileInfo> = HashSet::new();
-                if let Some(inode) = stat.inode {
-                    inodes.insert(inode);
+        if let Ok(stat) = Stat::new(path, &options) {
```

Comment:
Could you please add a comment explaining what this block is doing? It isn't super obvious reading the code

Reply:
Please see newest changes!